### ISSUE TITLE
Dependency Issues with Scipy and Pyaml-Include

### ISSUE THREAD

**Issue by Helw150** [stanford-crfm/levanter#543]
We've got a few dependency issues in our current setup that have popped up this week. (Of course while David is on vacation :laughing: - classic). Keeping track of these along with the workarounds I'm using here.

1) The most recent Scipy update is causing errors all the way down to Jax somehow - it's not a new major version so a bit confused at that one:
```
Traceback (most recent call last):
...                                                                                                                                                                                 
10/site-packages/optax/_src/utils.py", line 25, in <module>                                                                                                               
    import jax.scipy.stats.norm as multivariate_normal            
  File "/home/wheld3/venv310/lib/python3.10/site-packages/jax/scipy/stats/__init__.py", line 40, in <module>
    from jax._src.scipy.stats.kde import gaussian_kde as gaussian_kde                                                                                                                                              
  File "/home/wheld3/venv310/lib/python3.10/site-packages/jax/_src/scipy/stats/kde.py", line 26, in <module>         
    from jax.scipy import linalg, special                                                                                                                                                                          
  File "/home/wheld3/venv310/lib/python3.10/site-packages/jax/scipy/linalg.py", line 18, in <module>
    from jax._src.scipy.linalg import (                                                                  
  File "/home/wheld3/venv310/lib/python3.10/site-packages/jax/_src/scipy/linalg.py", line 402, in <module>
    @_wraps(scipy.linalg.tril)                                                                           
AttributeError: module 'scipy.linalg' has no attribute 'tril' 
```

Workaround: `gcloud compute tpus tpu-vm ssh tpu-vm --zone us-central2-b --worker=all --command "source venv310/bin/activate; pip install scipy==1.12.0`

2) The most recent Pyyaml-Include update is causing errors in Draccus, Filed an Issue to track there: https://github.com/dlwh/draccus/issues/12

Workaround: `gcloud compute tpus tpu-vm ssh tpu-vm --zone us-central2-b --worker=all --command "source venv310/bin/activate; pip install \"pyyaml-include<=1.4.1\""`

**Comment by Helw150** [comment]
I'm just working around these for now, since both are issues in underlying libraries. But I think it's also worth thinking about what level of version locking we should do on these?

**Comment by dlwh** [comment]
Yeah I mean. I wouldn’t have locked scipy since it’s transitive. But we
should probably lock things down a bit more.

I remember seeing a pypi time travel tool a week or two ago

**Comment by dlwh** [comment]
fixed in #545
